build/Makefile

```diff
@@ -95,38 +98,63 @@ system-keys : ${PDO_ENCLAVE_CODE_SIGN_PEM}
 ${PDO_ENCLAVE_CODE_SIGN_PEM} :
 	openssl genrsa -3 -out ${PDO_ENCLAVE_CODE_SIGN_PEM} 3072
 
-keys : $(PYTHON_DIR)
-	. $(abspath $(DSTDIR)/bin/activate) ; \
-		for i in 1 2 3 4 5 ; do $(KEYGEN) --keyfile $(DSTDIR)/opt/pdo/keys/eservice$${i} --format skf; done
-	. $(abspath $(DSTDIR)/bin/activate) ; \
-		for i in 1 2 3 4 5 ; do $(KEYGEN) --keyfile $(DSTDIR)/opt/pdo/keys/sservice$${i} --format pem; done
-	. $(abspath $(DSTDIR)/bin/activate) ; \
-		for i in 1 2 3 4 5 ; do $(KEYGEN) --keyfile $(DSTDIR)/opt/pdo/keys/pservice$${i} --format pem; done
-	. $(abspath $(DSTDIR)/bin/activate) ; \
-		for i in 1 2 3 4 5 6 7 8 9 10 ; do $(KEYGEN) --keyfile $(DSTDIR)/opt/pdo/keys/user$${i} --format pem; done
-
-conf : $(PYTHON_DIR)
-	@ echo Create configuration files from templates
+
+service_indexes := 1 2 3 4 5
+ESERVICE_SKF := $(addprefix $(KEYDIR),$(foreach i,$(service_indexes),eservice$(i)_private.skf))
+SSERVICE_PEM := $(addprefix $(KEYDIR),$(foreach i,$(service_indexes),sservice$(i)_private.pem))
+PSERVICE_PEM := $(addprefix $(KEYDIR),$(foreach i,$(service_indexes),pservice$(i)_private.pem))
+
+user_indexes := 1 2 3 4 5 6 7 8 9 10
+USER_PEM := $(addprefix $(KEYDIR),$(foreach i,$(user_indexes),user$(i)_private.pem))
+
+%.skf :
+	. $(abspath $(DSTDIR)/bin/activate) ; $(KEYGEN) --keyfile $(subst _private,,$*) --format skf
+
+%.pem :
+	. $(abspath $(DSTDIR)/bin/activate) ; $(KEYGEN) --keyfile $(subst _private,,$*) --format pem
+
+keys : $(ESERVICE_SKF) $(SSERVICE_PEM) $(PSERVICE_PEM) $(USER_PEM)
+
+ESERVICE_CONF := $(ETCDIR)/eservice1.toml
```

Comment:
currently makefile would not work as expected iff only some of the config files are missing but  x1.toml exists.  Doesn't seem a common error case, but couldn't you define the X_CONF here as you define the _SKF with the foreach in which case this would be a  complete non-issue?

Reply:
yes... but since we reconstruct all anyway, we only need one. that is... this is ugly, but works more or less exactly how it should.

Reply:
the main/only observationconcern is that right now if a single file (other than the first one) gets deleted or for whatever reason we decide to add a 6 file, then make would wrongly not regenerate the missing files. But definitely a blocker: the 'changes required' i only added because of the other breaking typo